ui/decredmaterial/editor.go

```diff
@@ -5,59 +5,69 @@ package decredmaterial
 import (
 	"image"
 	"image/color"
+	"strings"
 
 	"gioui.org/widget/material"
 
 	"gioui.org/f32"
 	"gioui.org/layout"
 	"gioui.org/op"
 	"gioui.org/op/paint"
+	"gioui.org/text"
 	"gioui.org/unit"
 	"gioui.org/widget"
 
 	"github.com/atotto/clipboard"
 	"golang.org/x/exp/shiny/materialdesign/icons"
+	"golang.org/x/image/math/fixed"
 )
 
 type Editor struct {
 	material.EditorStyle
 
-	TitleLabel Label
-	ErrorLabel Label
-	LineColor  color.RGBA
+	shaper           text.Shaper
+	StartHint        string
+	TitleLabel       Label
+	ErrorLabel       Label
+	LineColor        color.RGBA
+	LineColorFocused color.RGBA
 
 	flexWidth float32
 	//IsVisible if true, displays the paste and clear button.
 	IsVisible bool
 	//IsRequired if true, displays a required field text at the buttom of the editor.
 	IsRequired bool
-	//IsTitleLabel if true makes the title label visible.
-	IsTitleLabel bool
 
 	requiredErrorText string
 
 	pasteBtnMaterial IconButton
 	clearBtMaterial  IconButton
+
+	hasCalculated                                   bool
+	editorLineHeight                                int
+	titleLabelDims, editorDims, lineDims, errorDims layout.Dimensions
 }
 
-func (t *Theme) Editor(editor *widget.Editor, hint string) Editor {
+func (t *Theme) Editor(editor *widget.Editor, hint string) *Editor {
 	errorLabel := t.Caption("")
 	errorLabel.Color = t.Color.Danger
 
 	m := material.Editor(t.Base, editor, hint)
 	m.TextSize = t.TextSize
 	m.Color = t.Color.Text
-	m.Hint = hint
 	m.HintColor = t.Color.Hint
 
-	return Editor{
+	e := Editor{
 		EditorStyle:       m,
+		shaper:            t.Shaper,
 		TitleLabel:        t.Body2(""),
+		StartHint:         hint,
```

Comment:
Any reason why this is made public? Plus, I don't think a "StartHint" field is needed.